pages/add/add.jsx

```diff
@@ -190,32 +190,30 @@ export default React.createClass({
               </div>
             </div>);
   },
-  getFailurePrompt() {
-    return ( <HintMessage iconComponent={<span className={`fa fa-user`}></span>}
-                          header={`Sign in failed`}
-                          linkComponent={<Link to={`/featured`}>Explore featured</Link>}>
-              <p>Sorry, login failed! Please refresh and try again.</p>
-            </HintMessage>);
-  },
   getAnonymousContent() {
+    let header = `Please sign in to add a post`;
     let linkComponent = <a href={user.getLoginURL(utility.getCurrentURL())}
                            onClick={this.handleSignInBtnClick}>
                            Sign in with Google
                         </a>;
+    let additionalMessage;
+
+    if (user.failedLogin) {
+      header = `Sign in failed`;
+      linkComponent = <Link to={`/featured`}>Explore featured</Link>;
+      additionalMessage = <p>Sorry, login failed! Please try again or <a href="mailto:https://mzl.la/pulse-contact">contact us</a>.</p>;
+    }
 
-    return (<HintMessage iconComponent={<span className={`fa fa-user`}></span>}
-                         header={`Please sign in to add a post`}
+    return <HintMessage iconComponent={<span className={`fa fa-user`}></span>}
+                         header={header}
                          linkComponent={linkComponent}>
-            </HintMessage>);
+              {additionalMessage}
+            </HintMessage>;
   },
   getContent() {
-    if (user.loggedin) {
-      return this.getContentForLoggedInUser();
-    }
+    if (user.loggedin === undefined) return null;
```

Comment:
It takes a few seconds to find out if the current user is logged in or not. We shouldn't assume the current user is anonymous and show him/her "Please sign in" message. I changed the default value for `user.loggedin` to `null` (instead of `false`) so we can show a blank screen while getting result back from the `/userstatus` call.